modern/src/runtime/AnimatedLayer.ts

```diff
@@ -2,6 +2,97 @@ import Layer from "./Layer";
 import { unimplementedWarning } from "../utils";
 
 class AnimatedLayer extends Layer {
+  _playing: boolean;
+  _frameNum: number;
+
+  constructor(node, parent, annotations, store) {
+    super(node, parent, annotations, store);
+
+    if (!node.attributes.hasOwnProperty("autoplay")) {
+      node.attributes.autoplay = "0";
+    }
+
+    if (!node.attributes.hasOwnProperty("autoreplay")) {
+      node.attributes.autoreplay = "1";
+    }
+
+    if (!node.attributes.hasOwnProperty("speed")) {
+      node.attributes.speed = "200";
+    }
+
+    node.attributes.autoplay = !!Number(node.attributes.autoplay);
+    node.attributes.autoreplay = !!Number(node.attributes.autoreplay);
+
+    this._initializeStartEnd(node);
+
+    this._playing = node.attributes.autoplay;
+    this._frameNum = node.attributes.start || 0;
+
+    this.js_listen("js_framechange", () => {
+      this._frameNum += 1;
+      if (this._frameNum > Number(this.getendframe())) {
```

Comment:
`getEndFrame` should already return a number.

Reply:
AH yea. Got a bit confused as to where these conversions should actually be happening.

The values get read in from XML as strings, but then the `set` functions expect integers and the `get` functions return integers. I'm switching it to converting them to integers after filling in default values in the constructor.

thanks